libreyolo/models/rfdetr/keypoints.py

```
@@ -0,0 +1,373 @@
+        num_keypoints_per_class: Number of keypoints per class.
+
+    Returns:
+        Tuple ``(cost_l1, cost_findable, cost_visible, cost_nll)`` each of shape ``(B, Q, N)``.
+    """
+    b, num_queries, total_num_keypoints, pred_dim = all_pred_keypoints.shape
+    assert pred_dim >= 7, "Expected all_pred_keypoints last dim >= 7 (x,y,2 logits + 3 chol params)."
+    n_targets = target_keypoints.shape[0]
+    num_classes = len(num_keypoints_per_class)
+
+    if num_classes == 0:
+        raise ValueError("num_keypoints_per_class must be non-empty when computing keypoint matching costs.")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)